src/rest-server/src/controllers/v2/user.js

```diff
@@ -40,26 +52,33 @@ const getUser = async (req, res, next) => {
 const getAllUser = async (req, res, next) => {
   try {
     const userList = await userModel.getAllUser();
-    let retUserList = [];
-    for (let userItem of userList) {
-      userItem['admin'] = userItem.grouplist.includes(authConfig.groupConfig.adminGroup.groupname);
-      userItem['virtualCluster'] = userItem['extension']['virtualCluster'] ? userItem['extension']['virtualCluster'] : [];
-      delete userItem['password'];
-      retUserList.push(userItem);
+    const groupInfo = await groupModel.getAllGroup();
+    const allVClist = Object.keys(await vcModel.prototype.getVcListAsyc());
+    const groupMap = {};
+    for (const groupItem of groupInfo) {
+      groupMap[groupItem.groupname] = groupItem;
     }
+
+    const retUserList = await Promise.all(userList.map(async (userItem) => {
+      const groupItems = Array.from(userItem.grouplist, (groupname) => groupMap[groupname]);
+      const admin = groupModel.getAdminWithGroupinfo(groupItems);
+      userItem.admin = admin;
+      userItem.virtualCluster = admin ? allVClist : await groupModel.getVCsWithGroupinfo(groupItems);
+      delete userItem.password;
+      return userItem;
+    }));
     return res.status(200).json(retUserList);
   } catch (error) {
     return next(createError.unknown(error));
   }
 };
 
+// OIDC
```

Comment:
JSDOC is better

Reply:
Add jsdoc for specific funcs, the comments will be kept. Will mark all jsdoc comment resolved